### PR TITLE
Promotion pr labels

### PR DESCRIPTION
**Pull request by bitte-ein-bit**
## Description

This pull request introduces support for specifying custom labels for individual promotion PRs in the configuration, and ensures these labels are merged with top-level promotion PR labels without duplicates. The changes update the configuration schema, documentation, and implementation, and add comprehensive tests for the label merging logic.

### Configuration and Documentation Updates

* Added a new `labels` field to the `promotionPrs` configuration, allowing users to specify custom labels for each promotion PR. The documentation (`docs/installation.md`) and schema (`schema/telefonistka.json`) were updated to reflect this change. [[1]](diffhunk://#diff-8b042e3f94ca5c59a7cd990b950aec0073ea84fe811e7b22be51158d7b180d56R130) [[2]](diffhunk://#diff-e9c533475741aa4f4e2d1585ed7ebeea89d6b9484394fb4de043de9161764704R60-R66)
* Renamed the top-level configuration key from `promtionPRlables` to `promotionPrLabels` for clarity and consistency in docs, schema, and code. [[1]](diffhunk://#diff-8b042e3f94ca5c59a7cd990b950aec0073ea84fe811e7b22be51158d7b180d56R191-R192) [[2]](diffhunk://#diff-e9c533475741aa4f4e2d1585ed7ebeea89d6b9484394fb4de043de9161764704L89-R96) [[3]](diffhunk://#diff-feb81af7e47b8d0f0591ac857df4896bb89f167b694ce416cf6d5756640f7a11L40-R41)

### Implementation Changes

* Updated the `PromotionPr` struct to include the new `Labels` field, and ensured it is passed through the promotion planning logic. [[1]](diffhunk://#diff-feb81af7e47b8d0f0591ac857df4896bb89f167b694ce416cf6d5756640f7a11R26) [[2]](diffhunk://#diff-0a41799043384229949354e45f76bc9782955e2ce79964db03da188fcf5354b2R28) [[3]](diffhunk://#diff-0a41799043384229949354e45f76bc9782955e2ce79964db03da188fcf5354b2R235)
* Added a `mergeLabels` function to deduplicate and merge top-level and PR-specific labels, and integrated it into the PR creation flow. The `createPrObject` function now accepts a labels parameter and applies merged labels to PRs. [[1]](diffhunk://#diff-7026d4022cf2566206acc90b901d7e6b0dd6969e5ef2f862c963f6552b7becfeR673-R697) [[2]](diffhunk://#diff-7026d4022cf2566206acc90b901d7e6b0dd6969e5ef2f862c963f6552b7becfeL781-R809) [[3]](diffhunk://#diff-7026d4022cf2566206acc90b901d7e6b0dd6969e5ef2f862c963f6552b7becfeL1318-R1346) [[4]](diffhunk://#diff-7026d4022cf2566206acc90b901d7e6b0dd6969e5ef2f862c963f6552b7becfeL1336-R1374)

### Testing

* Added and updated unit tests to verify correct parsing of the new configuration options and the label merging logic, including edge cases for duplicate and overlapping labels. [[1]](diffhunk://#diff-4e880a8ba8689306a9c45b90230359714136473c7f98509b6d918a66b5e06913R39-R41) [[2]](diffhunk://#diff-4e880a8ba8689306a9c45b90230359714136473c7f98509b6d918a66b5e06913R60-R62) [[3]](diffhunk://#diff-4e880a8ba8689306a9c45b90230359714136473c7f98509b6d918a66b5e06913R83-R84) [[4]](diffhunk://#diff-d34eabe7c2435238517adacd977c66ec56021487bfd1f07e8a99c9e5c6313f99R10-R11) [[5]](diffhunk://#diff-d34eabe7c2435238517adacd977c66ec56021487bfd1f07e8a99c9e5c6313f99R20-R29) [[6]](diffhunk://#diff-efa540b5fb84825609760caa99afa47cef9362bae91d5522fb699c91c83c5cabR631-R699)

These changes make promotion PR labeling more flexible and robust, improving both configuration clarity and downstream automation.

## Type of Change

- [x] Bug Fix
- [x] New Feature
- [x] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/commercetools/telefonistka/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: internal/pkg/configuration/tests/testConfigurationParsing.yaml [hunk 1]
```diff
@@ -15,14 +17,16 @@ promotionPaths:
...
+        - "prod-us"
   - sourcePath: "env/prod/us-central1/c2/"
     conditions:
     promotionPrs:
       - targetPaths:
         - "env/prod/us-west1/c2/"
         - "env/prod/us-central1/c3/"
 
-promtionPrLables:
+promotionPrLabels:
```
Comment (hnnsgstfssn) [inline internal/pkg/configuration/tests/testConfigurationParsing.yaml]: This is a breaking change.

File: internal/pkg/configuration/config.go [hunk 2]
```diff
@@ -37,7 +38,7 @@ type Config struct {
 	PromotionPaths []PromotionPath `yaml:"promotionPaths"`
 
 	// Generic configuration
-	PromtionPrLables             []string               `yaml:"promtionPRlables"`
+	PromotionPrLabels            []string               `yaml:"promotionPrLabels"`
```
Comment (hnnsgstfssn) [inline internal/pkg/configuration/config.go]: This is a breaking change. The field name can be fixed, but the YAML tag needs to stay unfortunately.

**Comment by bitte-ein-bit** [comment]
@Freyert any updates or further questions?
